shuriken/include/shuriken/parser/Macho/macho_commands.h

```diff
@@ -0,0 +1,81 @@
+//--------------------------------------------------------------------*- C++ -*-
+// Shuriken-Analyzer: library for bytecode analysis.
+// @author Farenain <kunai.static.analysis@gmail.com>
+//
+// @file header.h
+// @brief MachoCommand of a MACHO file represented by a structure
+
+#ifndef SHURIKENLIB_MACHO_COMMANDS_H
+#define SHURIKENLIB_MACHO_COMMANDS_H
+
+#include "shuriken/parser/Macho/macho_sections.h"
+#include "shuriken/common/shurikenstream.h"
+
+#include <vector>
+#include <memory>
+
+namespace shuriken::parser::macho {
+    class MachoCommands {
+    public:
+        /// @brief Structure with the definition of the MACHO load command
+        /// all these values are later used for parsing the load command
+        /// from MACHO
+        struct loadcommand_t {
+            uint32_t cmd;               //! command type
+            uint32_t cmdsize;           //! command size
+        };
+
+        /// @brief Structure with the definition of the MACHO segment load command
+        /// all these values are later used for parsing the segment load command
+        /// from MACHO
+        struct segmentcommand_t {
+            uint32_t cmd;               //! command type
+            uint32_t cmdsize;           //! command size
+            char segname[16];           //! segment type
+            uint64_t vmaddr;            //! virtual memory address 
+            uint64_t vmsize;            //! size of the segment in the virtual memory
+            uint64_t fileoff;           //! offset 
+            uint64_t filesize;          //! size of the segment's data
+            uint32_t maxprot;           //! maximum VM protection 
+            uint32_t initprot;          //! initial VM protection 
+            uint32_t nsects;            //! number of sections 
+            uint32_t flags;             //! flags 
+        };
+
+        using loadcommands_t = std::vector<std::unique_ptr<loadcommand_t>>;
```

Comment:
create another ones to keep `std::vector<std::reference_wrapper<loadcommand_t>>`, in this way you don't have to return a vector of `unique_ptr`